proposals/span-safety.md

```diff
@@ -37,6 +37,57 @@ Note:
 *Possible incremental relaxations*
 - It may be possible to allow span locals be defined and used in a block inside an async method as long as the block does not contain `await` expressions. 
 
+## Generalized `ref-like` types in source code.
+
+`ref-like` structs will have to be explicitly marked in the source code using `ref` modifier:
+
+```c#
+ref struct TwoSpans<T>
+{
+	// can have ref-like instance fields
+	public Span<T> first;
+	public Span<T> second;
+} 
+
+// error: arrays of ref-like types are not allowed. 
+TwoSpans<T>[] arr = null;
+
+``` 
+
+Designating a struct as ref-like will allow the struct to have ref-like instance fields and will also make all the requirements of ref-like types applicable to the struct. 
```

Comment:
Will it somehow be possible to create a ref-like struct that contains `ref` fields?